modules/src/mock_context.rs

```diff
@@ -0,0 +1,199 @@
+use crate::ics02_client::client_def::{AnyClientState, AnyConsensusState};
+use crate::ics03_connection::context::{ConnectionKeeper, ConnectionReader};
+use crate::ics03_connection::error::Error as ICS3Error;
+use crate::ics23_commitment::commitment::CommitmentPrefix;
+use crate::ics24_host::identifier::{ClientId, ConnectionId};
+use std::collections::HashMap;
+use tendermint::block::Height;
+
+use crate::ics02_client::client_type::ClientType;
+use crate::ics02_client::context::ClientReader;
+use crate::ics03_connection::connection::ConnectionEnd;
+use crate::mock_client::header::MockHeader;
+use crate::mock_client::state::{MockClientRecord, MockClientState, MockConsensusState};
+use std::cmp::min;
+use std::convert::TryInto;
+
+/// Mock for a context. Used in testing handlers of all modules.
+#[derive(Clone, Debug)]
+pub struct MockContext {
+    /// Maximum size of the history.
+    max_history_size: usize,
+
+    /// Highest height of the headers in the history.
+    latest_height: Height,
+
+    /// A list of `max_history_size` headers, ascending order by their height (latest is last).
+    history: Vec<MockHeader>,
+
+    // All the connections in the store.
+    connections: HashMap<ConnectionId, ConnectionEnd>,
+
+    /// The set of all clients, indexed by their id.
+    clients: HashMap<ClientId, MockClientRecord>,
+
+    // Association between client ids and connection ids.
+    client_connections: HashMap<ClientId, ConnectionId>,
+}
+
```

Comment:
Could we keep the clients, connections, channels, packets,.. in the struct, init sequence, etc?
```suggestion
pub struct MockContext {
    /// Maximum size of the history.
    max_history_size: usize,

    /// Highest height of the headers in the history.
    latest_height: Height,

    /// A list of `max_history_size` headers, ascending order by their height (latest is last).
    history: Vec<MockHeader>,

    /// The set of all clients, indexed by their id.
    clients: HashMap<ClientId, MockClientRecord>,
    
    // Association between client ids and connection ids.
    client_connections: HashMap<ClientId, ConnectionId>,
    
    // All the connections in the store.
    connections: HashMap<ConnectionId, ConnectionEnd>,
}

```